.github/workflows/linter_actions.yml

```diff
@@ -10,9 +10,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - name: Run black linter
-        uses: psf/black@stable
-      # use the flake8 linter to annotate improperly formatted code
-      # note linter arguments are supplied via the .flake8 config file
-      - name: Annotate PR after running flake8
-        uses: TrueBrain/actions-flake8@v2
+
+      # Use the Ruff linter to annotate code style / best-practice issues
+      # NOTE: More config provided in pyproject.toml
+      - name: Lint and annotate PR
+        uses: chartboost/ruff-action@v1
```

Comment:
Looks like this GitHub Action isn't compatible with ruff>0.5.0 (https://github.com/ChartBoost/ruff-action/issues/30) and there's also a separate issue that they're looking into transferring this to another organization (which might take a while).

So... could we just use [`pre-commit.ci`](https://pre-commit.ci/) instead? Or are the annotations good to have?

Reply:
I'm cool with just using pre-commit.ci. I was trying to avoid any regression in functionality. I personally don't find annotation of PRs to be that valuable, but I imagine the feature would be more valuable to someone who doesn't know how to use the linter at the CLI. :thinking: 

It looks to me from reading the issue like this isn't an incompatibility issue _yet_ though? Or has Ruff gone forward with the breaking change?

Reply:
Fixed linting on this PR. I was using an outdated argument form!

Reply:
> I personally don't find annotation of PRs to be that valuable, but I imagine the feature would be more valuable to someone who doesn't know how to use the linter at the CLI.

You've identified exactly why we have the PR annotations set up. Precommit is an additional barrier to new contributors, so I wanted to be able to enforce the linting and formatting without absolutely requiring they complete yet another setup step to contribute.

Reply:
:100: